charts/supabase/templates/analytics/deployment.yaml

```diff
@@ -0,0 +1,154 @@
+{{- if .Values.analytics.enabled -}}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "supabase.analytics.fullname" . }}
+  labels:
+    {{- include "supabase.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.analytics.autoscaling.enabled }}
+  replicas: {{ .Values.analytics.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "supabase.analytics.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.analytics.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "supabase.analytics.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.analytics.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "supabase.analytics.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.analytics.podSecurityContext | nindent 8 }}
+      initContainers:
+        - name: init-db
+          image: postgres:15-alpine
+          imagePullPolicy: IfNotPresent
+          env:
+            - name: DB_HOST
```

Comment:
I'm a fan of how bitnami does these things, because it allows to easily refer to external dbs, e.g, https://github.com/bitnami/charts/blob/main/bitnami/supabase/templates/_helpers.tpl#L413C1-L426C12 . I am a n00b but is there a benefit to having `DB_HOST` as an explicit external env var for each submodule -> could there not be a `externalDatabase` like bitnami ( https://github.com/bitnami/charts/blob/main/bitnami/supabase/values.yaml#L3142)?

Reply:
I could add external db values somewhere under `db.external.host`. But as you can see, the main objective of this PR is to make sure all deployments are up-to-date with the official Supabase repo. Any QoL improvements may hopefully come after this PR is merged.

Reply:
Sure, have you had any signs of life from @arpagon though? It seems he might be too busy, and it might be better to just move forward and try and get things merged later. This already officially qualifies as a *super-mega-merge* PR, so a few extra large-scale changes probably wouldn't hurt :-). And `bitnami` are a common source of inspiration for charts around the interwebs. So we could continue this on your repo and then try and get that merged later maybe?